### DISCUSSION TITLE
A 'GrainBoundaryVelocity' is not a registered object

### DISCUSSION THREAD

**Discussion by mnabilbhuiyan** [idaholab/moose#16195] [Q&A General] [answered]
Hello,
I was trying to compute the grain boundary velocity. I followed the link
https://mooseframework.inl.gov/source/auxkernels/GrainBoundaryVelocity.html
I got the message:

 ***** ERROR ***
A 'GrainBoundaryVelocity' is not a registered object.

If you are trying to find this object in a dynamically linked library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.**

I need some suggestions how I can solve this issue.

Best Regards
Nabil

**Comment by aeslaughter** [comment, marked as answer]
Make sure you have the phase field module enabled. See "Enable Physics Modules" on this page: https://mooseframework.inl.gov/getting_started/new_users.html#create-an-app

**Comment by mnabilbhuiyan** [comment]
I enabled phase field module but it didn't work.

**Reply by aeslaughter** [reply]
Be sure to re-compile. What application are you running? If you run it with --registry do you see GrainBoundaryVelocity listed?

**Reply by mnabilbhuiyan** [reply]
My application name is eagle. You are mentioning about the following command to see GrainBoundaryVelocity?
mnabil@044476jh6xh03-u:~/projects/eagle$ ./eagle-opt --registry

**Reply by mnabilbhuiyan** [reply]
I am not seeing GrainBoundaryVelocity listed.

**Reply by aeslaughter** [reply]
Is Eagle hosted somewhere that I could take a look?

**Reply by mnabilbhuiyan** [reply]
All the applications are in my personal computer. Is there anyway that I can show you my applications?

**Reply by mnabilbhuiyan** [reply]
I have compiled some new applications. But I not seeing the GrainBoundaryVelocity and also getting the same error message
***** ERROR ***
A 'GrainBoundaryVelocity' is not a registered object.

**Reply by aeslaughter** [reply]
The best way to share the code would be to put it on Github
